### PR TITLE
prototype jax with ddpg

### PR DESCRIPTION
**Pull request by vwxyzjn**
## Types of changes

- [x] New algorithm

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/vwxyzjn/cleanrl/blob/master/CONTRIBUTING.md) guide (**required**).
- [x] I have ensured `pre-commit run --all-files` passes (**required**).
- [ ] I have updated the documentation and previewed the changes via `mkdocs serve`.
- [ ] I have updated the tests accordingly (if applicable).

If you are adding new algorithms or your change could result in performance difference, you may need to (re-)run tracked experiments. See https://github.com/vwxyzjn/cleanrl/pull/137 as an example PR. 
- [x] I have contacted @vwxyzjn to obtain access to the [openrlbenchmark W&B team](https://wandb.ai/openrlbenchmark) (**required**).
- [x] I have tracked applicable experiments in [openrlbenchmark/cleanrl](https://wandb.ai/openrlbenchmark/cleanrl) with `--capture-video` flag toggled on (**required**).
- [x] I have added additional documentation and previewed the changes via `mkdocs serve`.
    - [x] I have created a table comparing my results against those from reputable sources (i.e., the original paper or other reference implementation).
    - [x] I have added the learning curves (in PNG format with `width=500` and `height=300`).
    - [x] I have added links to the tracked experiments.
- [ ] I have updated the tests accordingly (if applicable).

### CODE REVIEW

**Comment by vwxyzjn** [comment]
@dosssman @huxiao09 I seem to have gotten CleanRL's DDPG + Jax working: about 5x speed up for free.

image

**Comment by vwxyzjn** [comment]
@dosssman and @ikostrikov could you help review this, please? I am unfamiliar with JAX so might be coding up things wrong or have really bad format...

**Comment by ikostrikov** [comment]
Looks good to me! The only thing I would add is TrainState:
https://flax.readthedocs.io/en/latest/flax.training.html#flax.training.train_state.TrainState

**Comment by vwxyzjn** [comment]
```
def update_target(src, dst, tau):
    return jax.tree_map(lambda p, tp: p * tau + tp * (1 - tau), src, dst)
```

is exactly the same as `optax.incremental_update`. cc @ikostrikov :)

**Comment by vwxyzjn** [comment]
@dosssman @yooceii  could you give a review, please? The changes have been finalized.

File: cleanrl/ddpg_continuous_action_jax.py
Comment (dosssman) [inline cleanrl/ddpg_continuous_action_jax.py:191]: Probably no need to clip here, since there is no noise addition, and the actor returns actions within the actinp space bounds.

**Review by dosssman — APPROVED** [review]
Not very familiar with Jax, so can't really suggest any quality improvement.
Beside that, its relatively easy to understand, and the algorithm logic looks good to me.
Great work as always.

**Comment by vwxyzjn** [comment]
@dosssman, @yooceii, and @joaogui1 this is ready for review with docs (https://cleanrl-git-jax-ddpg-vwxyzjn.vercel.app/rl-algorithms/ddpg/#ddpg_continuous_action_jaxpy, note some of the links don't work until this PR is merged).

File: cleanrl/ddpg_continuous_action_jax.py
Comment (joaogui1) [inline cleanrl/ddpg_continuous_action_jax.py:109]: This should be `x = x * self.action_scale + self.action_bias`, no?

Comment (vwxyzjn) [reply cleanrl/ddpg_continuous_action_jax.py]: Thanks for this great catch! I am fixing this and will merge after CI passes.

**Pull request merged**
